cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -479,3 +490,54 @@ func isAutoProvisioning(spc *apis.StoragePoolClaim) bool {
 func isManualProvisioning(spc *apis.StoragePoolClaim) bool {
 	return spc.Spec.BlockDevices.BlockDeviceList != nil
 }
+
+func (c *Controller) upgrade(spc *apis.StoragePoolClaim) (*apis.StoragePoolClaim, error) {
+	if spc.VersionDetails.Current != spc.VersionDetails.Desired &&
+		spc.VersionDetails.Current != "" && spc.VersionDetails.Desired != "" {
+		// As no other steps are required just change current version to
+		// desired version
+		spc.VersionDetails.Current = spc.VersionDetails.Desired
+		obj, err := c.clientset.OpenebsV1alpha1().StoragePoolClaims().Update(spc)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to update storagepoolclaim")
+		}
+		return obj, nil
+	}
+	return spc, nil
+}
+
+// populateVersion assigns VersionDetails for old spc object
+func (c *Controller) populateVersion(spc *apis.StoragePoolClaim) (*apis.StoragePoolClaim, error) {
+	if spc.VersionDetails.Current == "" {
+		var v string
+		cspList, err := c.clientset.OpenebsV1alpha1().CStorPools().List(
+			metav1.ListOptions{
+				LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spc.Name,
+			})
+		if err != nil {
+			return nil, errors.Wrapf(
+				err,
+				"failed to get csplist for %s to add version details",
+				spc.Name,
+			)
+		}
+		if len(cspList.Items) == 0 {
+			v = version.Current()
+		} else {
+			v = cspList.Items[0].Labels[string(apis.OpenEBSVersionKey)]
+		}
+		spc.VersionDetails.Current = v
+		// As there are no upgrade steps for spc in 1.3.0 and for newly created spc
```

Comment:
I don't think this is not correct `As there are no upgrade steps for spc in 1.3.0`

Reply:
Removed